bloom/xxhash/sum64uint_test.go

```diff
@@ -180,105 +181,80 @@ func reportThroughput(b *testing.B, loops, count int, start time.Time) {
-			start := time.Now()
-			for i := 0; i < b.N; i++ {
-				_ = xxhash.MultiSum64Uint16(out, in)
-			}
-			reportThroughput(b, b.N, len(out), start)
-		})
+	in := make([]uint16, benchmarkBufferSize/2)
+	for i := range in {
+		in[i] = uint16(i)
 	}
+	b.Run(fmt.Sprintf("%dKB", benchmarkBufferSize/1024), func(b *testing.B) {
+		out := make([]uint64, len(in))
+		start := time.Now()
+		for i := 0; i < b.N; i++ {
+			_ = xxhash.MultiSum64Uint16(out, in)
+		}
+		reportThroughput(b, b.N, len(out), start)
+	})
 }
 
 func BenchmarkMultiSum64Uint32(b *testing.B) {
-	for _, bb := range benchmarks {
-		if bb.n < 4 {
-			continue
-		}
-		in := make([]uint32, bb.n/4)
-		for i := range in {
-			in[i] = uint32(i)
-		}
-		out := make([]uint64, len(in))
-		b.Run(bb.name, func(b *testing.B) {
-			start := time.Now()
-			for i := 0; i < b.N; i++ {
-				_ = xxhash.MultiSum64Uint32(out, in)
-			}
-			reportThroughput(b, b.N, len(out), start)
-		})
+	in := make([]uint32, benchmarkBufferSize/4)
+	for i := range in {
+		in[i] = uint32(i)
 	}
+	b.Run(fmt.Sprintf("%dKB", benchmarkBufferSize/1024), func(b *testing.B) {
+		out := make([]uint64, len(in))
+		start := time.Now()
+		for i := 0; i < b.N; i++ {
+			_ = xxhash.MultiSum64Uint32(out, in)
+		}
+		reportThroughput(b, b.N, len(out), start)
+	})
 }
 
 func BenchmarkMultiSum64Uint64(b *testing.B) {
-	for _, bb := range benchmarks {
-		if bb.n < 16 {
-			continue
-		}
-		in := make([]uint64, bb.n/8)
-		for i := range in {
-			in[i] = uint64(i)
-		}
-		out := make([]uint64, len(in))
-		b.Run(bb.name, func(b *testing.B) {
-			start := time.Now()
-			for i := 0; i < b.N; i++ {
-				_ = xxhash.MultiSum64Uint64(out, in)
-			}
-			reportThroughput(b, b.N, len(out), start)
-		})
+	in := make([]uint64, benchmarkBufferSize/8)
+	for i := range in {
+		in[i] = uint64(i)
 	}
+	b.Run(fmt.Sprintf("%dKB", benchmarkBufferSize/1024), func(b *testing.B) {
+		out := make([]uint64, len(in))
+		start := time.Now()
+		for i := 0; i < b.N; i++ {
+			_ = xxhash.MultiSum64Uint64(out, in)
+		}
+		reportThroughput(b, b.N, len(out), start)
+	})
 }
 
 func BenchmarkMultiSum64Uint128(b *testing.B) {
-	for _, bb := range benchmarks {
-		if bb.n < 16 {
-			continue
-		}
-		in := make([][16]byte, bb.n/16)
-		for i := range in {
-			binary.LittleEndian.PutUint64(in[i][:8], uint64(i))
-			binary.LittleEndian.PutUint64(in[i][8:], uint64(i))
-		}
-		out := make([]uint64, len(in))
-		b.Run(bb.name, func(b *testing.B) {
-			start := time.Now()
-			for i := 0; i < b.N; i++ {
-				_ = xxhash.MultiSum64Uint128(out, in)
-			}
-			reportThroughput(b, b.N, len(out), start)
-		})
+	in := make([][16]byte, benchmarkBufferSize/16)
+	for i := range in {
+		binary.LittleEndian.PutUint64(in[i][:8], uint64(i))
+		binary.LittleEndian.PutUint64(in[i][8:], uint64(i))
 	}
+	b.Run(fmt.Sprintf("%dKB", benchmarkBufferSize/1024), func(b *testing.B) {
+		out := make([]uint64, len(in))
+		start := time.Now()
+		for i := 0; i < b.N; i++ {
+			_ = xxhash.MultiSum64Uint128(out, in)
```

Comment:
You need to actually assign this to a global sink variable right? Or isn't there a chance it will get optimized away?

Reply:
Or, oh, is out being written to every time, so it's fine

Reply:
I don't think you need to, the function is not implemented in Go so the compiler cannot inline it; this is a leftover from copying https://github.com/cespare/xxhash, maybe some linter required it 🤷

Reply:
I mean if you had

```
for i := 0; i < 5; i++ {
    _ = compute1000thFibonacciNumber()
}
```

in theory that could be entirely elided because it's a pure function and you do nothing with the output, is my understanding of how the compiler works.

Reply:
Go assumes it knows nothing about what's going on in the assembly functions, that's also why you need `//go:noescape` to help escape analysis know what the inputs aren't going to be retained.